BaragonService/src/main/java/com/hubspot/baragon/service/resources/LoadBalancerResource.java

```diff
@@ -60,7 +61,7 @@ public LoadBalancerResource(BaragonLoadBalancerDatastore loadBalancerDatastore,
 
   @POST
   @Path("/{clusterName}/sources")
-  public BaragonGroup addSource(@PathParam("clusterName") String clusterName, @QueryParam("source") String source) {
+  public BaragonGroup addSource(@PathParam("clusterName") String clusterName, BaragonSource source) {
```

Comment:
one option to maintain backwards compatibility (for a release -- not suggesting we do this forever) would be to accept both an optional query param and an optional JSON blob and throw a HTTP 400 if both or neither are supplied.